chrome_extension/content.js

```diff
@@ -0,0 +1,46 @@
+// The extension id assigned by Chrome to this extension.
+const extension_id = 'nimjbbjddomejdcjaobokgnfmjgjefcc';
+
+// Listen for messages from the top frame of the current tab.
+//
+// This may be sent with something similar to the following:
+//
+// ```
+// window.postMessage('oak_enter', '*');
+// ```
+//
+// Once this content script receives such message, then it sends in turn a message to the background
+// script of the extension.
+window.addEventListener(
+  'message',
+  (event) => {
+    console.log('message received', event);
+    if (event.data == 'oak_enter') {
+      chrome.runtime.sendMessage(extension_id, 'oak_enter');
+    }
+  },
+  false
+);
+
+// Injects a script in the context of the active tab, so that it can operate on the `window` object
+// of that tab.
+function inject(fn) {
+  const script = document.createElement('script');
+  script.text = `(${fn.toString()})();`;
+  document.documentElement.appendChild(script);
+}
```

Comment:
Can we use `chrome.tabs.executeScript` instead?
https://developer.chrome.com/extensions/tabs#method-executeScript

Injecting scripts like this could lead to unexpected consequences, by polluting the global JavaScript namespace of that page.

Reply:
Not needed any more.